include/zephyr/net/conn_mgr_monitor.h

```diff
@@ -84,6 +88,55 @@ void conn_mgr_ignore_l2(const struct net_l2 *l2);
  */
 void conn_mgr_watch_l2(const struct net_l2 *l2);
 
+/**
+ * @typedef net_conn_mgr_online_checker_t
+ * @brief Handler function that is called when network stack needs
+ * to do a HTTPS request and needs to ask application to setup TLS
+ * credentials and return them to the caller.
+ *
+ * @param iface Network interface where the online check is sent.
+ * @param sec_tag_list Array of TLS security tags returned to the connection manager.
+ * @param sec_tag_size Size of the returned tags array.
+ * @param tls_hostname User needs to set this and it is returned to the connection
+ *        manager. The value is used to set TLS_HOSTNAME socket option. If the value
+ *        is empty, the TLS connection most like will not work. The returned pointer
+ *        should point to a NULL terminated buffer that is valid when the callback
+ *        returns.
+ * @param url URL where the HTTPS request is sent.
+ * @param host Hostname where the HTTPS request is sent.
+ * @param port Port where the HTTPS request is sent.
+ * @param dst Destination address where the HTTPS request is sent.
+ * @param user_data A valid pointer to user data or NULL
+ *
+ * @retval <0 if this online check request is cancelled and not done.
+ * @retval  0 the online check is done by the connection manager.
+ */
+typedef int (*net_conn_mgr_online_checker_t)(struct net_if *iface,
+					     const sec_tag_t **sec_tag_list,
+					     size_t *sec_tag_size,
+					     const char **tls_hostname,
+					     const char *url,
+					     const char *host,
+					     const char *port,
+					     struct sockaddr *dst,
+					     void *user_data);
```

Comment:
I think it is very useful to be able to configure things at runtime, but I also see some benefits with build time configuration. I think we could let this decision to end user, so we could have a Kconfig option that allows both ways.

Reply:
My main motivation for switching to build-time is that it would significantly reduce the complexity of certain aspects of this contribution, so if we do it both ways, we don't get that complexity reduction

Reply:
Sorry but I do not see the complexity here.